classes/Link.php

```diff
@@ -503,25 +507,25 @@ public function getAdminLink($controller, $with_token = true, $sfRouteParams = a
                 if (!$redirectLegacy) {
                     if (array_key_exists('id_product', $sfRouteParams)) {
                         if (array_key_exists('deleteproduct', $sfRouteParams)) {
-                            return $this->getBaseLink().basename(_PS_ADMIN_DIR_).'/product/unit/delete/' . $sfRouteParams['id_product'];
+                            return $this->getBaseLink().basename(_PS_ADMIN_DIR_).$index.'/product/unit/delete/'.$sfRouteParams['id_product'];
```

Comment:
you can use:

```php
$kernel->getContainer()->get('router')->generate('admin_product_unit_action', array('action' => 'delete', 'id' => 10), true);
```

Pros:
* no more hardcoded urls here
* absolute urls
* index.php

Cons:
* index.php will always be "printed" : is that an issue ?
* Link class now depends on Symfony router